### PR TITLE
[FEATURE] add function to quickly export MS2 peaks in rt,mz,int format

### PR DESCRIPTION
**Pull request by jpfeuffer**
For usage in pyopenms

### CODE REVIEW

File: src/openms/include/OpenMS/KERNEL/MSExperiment.h [hunk 1]
```diff
@@ -377,6 +377,28 @@ namespace OpenMS
       }
     }
 
+    // for fast pyOpenMS access to MS2 peak data in format: [rt, mz, intensity]
+    void getMS2PeakData(
```
Comment (timosachsenberg) [inline src/openms/include/OpenMS/KERNEL/MSExperiment.h:381]: a level parameter could make sense here as we export all peaks (as opposed to the MS1 only area iterator version above)

**Comment by timosachsenberg** [comment]
related to #5722

File: src/openms/include/OpenMS/KERNEL/MSExperiment.h [hunk 2]
```diff
@@ -377,6 +377,28 @@ namespace OpenMS
...
+      std::vector<float>& rt, 
+      std::vector<float>& mz, 
+      std::vector<float>& intensity) const
+    {
+      for (const auto& s : spectra_)
+      {
+        if (s.getMSLevel() == 2)
+        {
+          rt.reserve(rt.size() + s.size());
```
Comment (cbielow) [inline src/openms/include/OpenMS/KERNEL/MSExperiment.h:390]: these reserves are probably a pessimization

**Comment by jpfeuffer** [comment]
@timosachsenberg is this still needed? I guess not. @axelwalter was happy with the current speed.
If we continue we need to discuss the API of this function. this is just a suggestion.

**Comment by timosachsenberg** [comment]
not sure... probably nice to have. not high priority though

**Pull request closed without merging**